packages/core/src/services/documentLogs/fetchDocumentLogsWithEvaluationResults.ts

```diff
@@ -81,3 +89,49 @@ export async function fetchDocumentLogsWithEvaluationResults(
     }
   })
 }
+
+export async function findDocumentLogWithEvaluationResultPage(
+  {
+    workspaceId,
+    documentUuid,
+    commit,
+    documentLogId,
+    pageSize = DEFAULT_PAGE_SIZE,
+  }: {
+    workspaceId: number
+    documentUuid: string
+    commit: Commit
+    documentLogId: string
+    pageSize?: string
+  },
+  db = database,
+) {
+  const documentLogsRepo = new DocumentLogsRepository(workspaceId, db)
+  const result = await documentLogsRepo.find(Number(documentLogId))
+  if (result.error) return undefined
+  const log = result.value
+
+  const position = (
+    await db
+      .select({
+        count: sql`count(*)`.mapWith(Number).as('count'),
+      })
+      .from(documentLogs)
+      .innerJoin(
+        commits,
+        and(eq(commits.id, documentLogs.commitId), isNull(commits.deletedAt)),
+      )
+      .where(
+        and(
+          eq(documentLogs.documentUuid, documentUuid),
+          getCommitFilter(commit),
+          sql`(${documentLogs.createdAt}, ${documentLogs.id}) >= (${new Date(log.createdAt).toISOString()}, ${log.id})`,
+        ),
```

Comment:
is there any perf benefits of doing the combined equality check? otherwise i find `and(gte(log.createdAt, documentLogs.createdAt), eq(log.id, documentLogs.id)` much easier to read

Reply:
This is a Pair comparison. So it would be more like, if `(documentLogs.createdAt > log.createdAt) or (documentLogs.createdAt = log.createdAt) and documentLogs.id > log.id)`. I think I prefer to do a simple Pair comparison, much more readable imo